json_yaml_files/prepare_files.py

```diff
@@ -4,17 +4,22 @@
 import json
 import sys
 import yaml
+import argparse
 from pathlib import Path
 
 
 class BIDStoNDAConfigGenerator:
     def __init__(self):
-        self.path = sys.argv[1]
-        self.layout = BIDSLayout(self.path, derivatives=True)
+        # fetch the 
+        parser = argparse.ArgumentParser(description='Process a dataset path.')
```

Comment:
A more verbose description would be good. Maybe this?

```suggestion
        parser = argparse.ArgumentParser(description='Prepare JSON and YAML files for a single dataset path.')
```